lib/Simple-Circular-Queue/library.json

```diff
@@ -0,0 +1,11 @@
+{
```

Comment:
Why were Simple Circular Queue and Tiny USB copied into this repo? Did you have to make changes to them?

Reply:
Hi, do you mind if I rewrite the code to be like https://github.com/ZuluSCSI/ZuluSCSI-firmware. where the general flow is in the platformio src directly and the mcu specific code is in the lib directory?

I didn't know we would be sending a pull request to your repo at the time of getting it to work with the RP2040

Reply:
Do you mean where there is just a "main.cpp" in the src directory and all of the other files are in the lib dir? If so, I fully support that.

I would like to keep as much of the source common as possible. In Zulu, it looks like there are several copies of scsiPhy.cpp (as an example). I think that adds a lot of maintenance overhead.

The only major differences between the RP2040 and the stm32 should be the GPIO interface. QuakkADB and adbuino are both using TinyUSB, so that code should be extremely similar.

Proposed directory structure:
```
/case
/doc
/hw
/images
/src
	/adbuino (we can rename this to something common if you want, like "firmware")
		/boards
			QuokkADB_RP2040.json
		/include
			/pico
				config_autogen.h
			/atmelavr
				anything_atmel_specific.h
			adb.h
			adbregisters.h
			bithacks.
			pgmstrings.h
			ps2kbd.h
			usb_description.h
			usb_hid_keys.h
			usbinterface
		/lib
			/atmelavr
				atmelavr_specific.cpp (if there is anything)
			/rp2040
				rp2040_specific.cpp (if there is anything)
			adb.cpp
			adbuino.cpp
			ps2kbd.cpp
			usb_description.cpp
			usbinterface.cpp
			usbkdbparser.cpp
			usbmouseparser.cpp
			usbtoadb.cpp
		/src
			main.cpp
		/test
		platform.ini
	/adbmon (we could go for a similar structure as under adbuino here, if you want. I don't expect this will be useful to many people except developers... and even then, its not that great.)
```